### DISCUSSION TITLE
[ZK Stack] - What is the difference between blobs and calldata in the code?

### DISCUSSION THREAD

**Discussion by Marius-BC-Dev** [zkSync-Community-Hub/zksync-developers#392] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.73.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

What is the difference between blobs and calldata in the code? Is this one saving gas fee?

### Expected Behavior

Reduction of gas fee

### Current Behavior

Not tested at the moment, not sure what blobs do.

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by jillalbritton** [comment]
In Ethereum smart contract development, "calldata" refers to the input data passed to a function during contract execution, while "blobs" typically refer to arbitrary data stored off-chain or in storage within the contract. Blobs are usually used for larger or more complex data that is not passed directly as function arguments. The use of blobs versus calldata can affect gas fees, but it depends on how they are utilized within the contract.

**Comment by syringjerry** [comment]
You can interact with smart contracts on zkSync using the Python SDK by deploying contracts, calling their functions, and querying their states. You may refer to the zkSync Python SDK documentation and explore examples provided there for guidance.

**Comment by bxpana** [comment]
@Marius-BC-Dev is there a specific part of the code you were asking about or just in general asking what is the difference between blobs and calldata?
